app/src/main/java/tech/relaycorp/courier/domain/PublicSync.kt

```diff
@@ -29,14 +30,17 @@ class PublicSync
     }
 
     private suspend fun syncUnhandled() {
-        state.send(State.DeliveringCargo)
-        cargoDelivery.deliver()
+        val dohClient = DoHClient()
+        dohClient.use {
```

Comment:
This is the main reason why I didn't integrate `DohClient` in the CogRPC library: The DoH client should be reused across CogRPC connections, so that we can leverage caching and reuse the connection with the DNS resolver where possible.

Another reason is that this extra DNS lookup is only needed by the courier app.

Reply:
It could have a module there for reusing it, but I'm ok with this option. Unless there would be many other things like these in the future, and we start having a train of dependencies to pass on each method 😛

Reply:
I _think_ this will be the only place where we ever need to resolve DNS records in this app, which is another reason why I chose a specific solution instead of a more generic one -- to keep the number of moving parts to a minimum. But we can always implement a more generic solution if it turns out we need it in more places.